shared/output/cloud/compute_small/lib/google/cloud/compute/v1/addresses/rest/client.rb

```diff
@@ -0,0 +1,516 @@
+              #
+              #     You can also filter nested fields. For example, you could specify scheduling.automaticRestart = false to include instances only if they are not scheduled for automatic restarts. You can use filtering on nested fields to filter based on resource labels.
+              #
+              #     To filter on multiple expressions, provide each separate expression within parentheses. For example, (scheduling.automaticRestart = true) (cpuPlatform = "Intel Skylake"). By default, each expression is an AND expression. However, you can include AND and OR expressions explicitly. For example, (cpuPlatform = "Intel Skylake") OR (cpuPlatform = "Intel Broadwell") AND (scheduling.automaticRestart = true).
+              #   @param max_results [::Integer]
+              #     The maximum number of results per page that should be returned. If the number of available results is larger than maxResults, Compute Engine returns a nextPageToken that can be used to get the next page of results in subsequent list requests. Acceptable values are 0 to 500, inclusive. (Default: 500)
+              #   @param order_by [::String]
+              #     Sorts list results by a certain order. By default, results are returned in alphanumerical order based on the resource name.
+              #
+              #     You can also sort results in descending order based on the creation timestamp using orderBy="creationTimestamp desc". This sorts results based on the creationTimestamp field in reverse chronological order (newest result first). Use this to sort resources like operations so that the newest operation is returned first.
+              #
+              #     Currently, only sorting by name or creationTimestamp desc is supported.
+              #   @param page_token [::String]
+              #     Specifies a page token to use. Set pageToken to the nextPageToken returned by a previous list request to get the next page of results.
+              #   @param project [::String]
+              #     Project ID for this request.
+              #   @param region [::String]
+              #     Name of the region for this request.
+              #
+              # @yield [result, env] Access the result along with the Faraday environment object
+              # @yieldparam result [::Google::Cloud::Compute::V1::AddressList]
+              # @yieldparam env [::Faraday::Env]
+              #
+              # @return [::Google::Cloud::Compute::V1::AddressList]
+              #
+              # @raise [::Faraday::Error] if the REST call is aborted.
+              def list request, options = nil
+                raise ::ArgumentError, "request must be provided" if request.nil?
+
+                request = ::Gapic::Protobuf.coerce request, to: ::Google::Cloud::Compute::V1::ListAddressesRequest
+
+                @addresses_stub.list request, options: options do |response, env|
+                  yield response, env if block_given?
+                end
```

Comment:
What happens to successful calls that result in 4xx or 5xx responses? I think the gRPC side maps those to `Google::Cloud::Error` exceptions, but I'm not convinced the above code does the same.

Reply:
Faraday's error handling middleware wraps it into the Faraday::Error of the appropriate type and then it is going to be caught and wrapped to Google::Cloud::Error (for Cloud)